Source/Particles/Deposition/CurrentDeposition.H

```diff
@@ -583,32 +583,50 @@ void doEsirkepovDepositionShapeN (const GetParticlePosition& GetPosition,
 
 #if defined(WARPX_DIM_3D)
 
-            for (int k=dkl; k<=depos_order+2-dku; k++) {
-                for (int j=djl; j<=depos_order+2-dju; j++) {
+            for (int k=0; k<=depos_order+2; k++) {
```

Comment:
We should try to throw in `#pragma unroll` directives for the loops:
https://developer.nvidia.com/blog/fast-dynamic-indexing-private-arrays-cuda/